src/engine/specification.md

```diff
@@ -197,25 +223,35 @@ This structure contains the attributes required to initiate a payload build proc
 #### Response
 
 * result: `object`
-    - `status`: `enum` - `"SUCCESS" | "SYNCING"`
+    - `status`: `enum` - `"VALID" | "INVALID" | "SYNCING" | "ACCEPTED"`
+    - `latestValidHash`: `DATA|null`, 32 Bytes - the hash of the most recent *valid* block in the branch defined by payload and its ancestors
+    - `validationError`: `String|null` - a message providing additional details on the validation error if the payload is deemed `INVALID`
     - `payloadId`: `DATA|null`, 8 Bytes - identifier of the payload build process or `null`
-* error: code and message set in case an exception happens while updating the forkchoice or initiating the payload build process.
+* error: code and message set in case an exception happens while validating payload, updating the forkchoice or initiating the payload build process.
 
 #### Specification
 
-1. The values `(forkchoiceState.headBlockHash, forkchoiceState.finalizedBlockHash)` of this method call map on the `POS_FORKCHOICE_UPDATED` event of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification) and **MUST** be processed according to the specification defined in the EIP.
+1. Client software **SHOULD** initiate the sync process if `forkchoiceState.headBlockHash` references an unknown payload or a payload that can't be validated because requisite data is missing. The sync process is specified in the [Sync process](#sync-process) section. Additionally:
+    * Client software **MUST** return `{status: SYNCING, latestValidHash: null, validationError: null, payloadId: null}` if the sync process has been initiated.
 
-2. All updates to the forkchoice state resulting from this call **MUST** be made atomically.
+1. Client software **MAY** skip an update of the forkchoice state and **MAY NOT** begin a payload build process if the payload identified by `forkchoiceState.headBlockHash` hasn't been validated yet. Additionally:
+    * Client software **MUST** return `{status: ACCEPTED, latestValidHash: null, validationError: null, payloadId: null}` in this case.
 
-3. Client software **MUST** return `{status: SUCCESS, payloadId: null}` if `payloadAttributes` is `null` and the client is not `SYNCING`.
+1. Client software **MAY** skip an update of the forkchoice state and **MAY NOT** begin a payload build process if `forkchoiceState.headBlockHash` references an ancestor of the head of the canonical chain. Additionally:
```

Comment:
Should this be a **MUST**? and **MUST NOT**?

Otherwise, it leaves open the option to use this API to (if supported by EL) rewind the fork choice and also build old payloads. The former (rewinding) might *only* be useful to user APIs but seems an abuse of this API. And the latter (building old payloads) might only really be used for attacks and other nuisance messages.

Reply:
Another option is to **MAY** update and **MUST NOT** begin build process. I think that restricting rewinding the head is a bit awkward.

Reply:
okay, that seems generally fine but see my comment about block leaves below